code/modules/power/supermatter/supermatter.dm

```diff
@@ -678,6 +678,19 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 		if(produces_gas)
 			env.merge(removed)
 			air_update_turf()
+	for(var/turf/turf_to_check in view(1, loc))
+		if(isspaceturf(turf_to_check))
+			var/integrity = get_integrity()
+			if(integrity < 15)
+				damage += max((power * 0.05) * DAMAGE_INCREASE_MULTIPLIER, 0.1)
+			else if(integrity < 30)
+				damage += max((power * 0.06) * DAMAGE_INCREASE_MULTIPLIER, 0.1)
+			else if(integrity < 50)
+				damage += max((power * 0.07) * DAMAGE_INCREASE_MULTIPLIER, 0.1)
+			else if(integrity < 75)
+				damage += max((power * 0.08) * DAMAGE_INCREASE_MULTIPLIER, 0.1)
+			else
+				damage += max((power * 0.09) * DAMAGE_INCREASE_MULTIPLIER, 0.1)
```

Comment:
This should be done in the portion of the sm that deals damage, you know the one.
As for the math itself, with a standard 1500 eer sm assuming the lowest integrity and one space tile, this code will deal 18.75 damage per tick, which is 10x the max damage the sm can take per tick, nix bullets.

I understand the logic behind this, but I think it should be done differently, perhaps with pressure, and not nearly as aggressively

Reply:
i changed it while you were writing this

Reply:
Ah redid the math, I used power * 0.05 by mistake. 0.0001 seems quite low, ya want and nerfed the thing when I wasn't looking